explorer/syntax/parser.ypp

```diff
@@ -1214,7 +1218,15 @@ declaration:
       $$ = arena->New<ConstraintDeclaration>(arena, context.source_loc(), $2,
                                              $3, $5);
     }
-| impl_kind_intro impl_deduced_params impl_type AS type_or_where_expression LEFT_CURLY_BRACE impl_body RIGHT_CURLY_BRACE
+| impl_declaration
+    { $$ = $1; }
+| match_first_declaration
+    { $$ = $1; }
+| alias_declaration
+    { $$ = $1; }
+;
+impl_declaration:
+  impl_kind_intro impl_deduced_params impl_type AS type_or_where_expression LEFT_CURLY_BRACE impl_body RIGHT_CURLY_BRACE
```

Comment:
Do you mean to support non-external declarations in match_first? You might want to handle this a little differently, because impl_kind_intro here allows it. You might want to add a test for it, to ensure it works correctly (e.g., if not supported, a compile error).

Reply:
`match_first` is only permitted at the top level, and in principle non-external declarations are only permitted within a class. But we don't actually enforce that `external` is used properly right now. I'll send a separate PR for that, then add tests here once it lands.

Reply:
Separate PR works for me.